### PR TITLE
feat: initial version

### PR DESCRIPTION
**Pull request by gberenice**
## what

- This adds the first version of the module for [Spacelift AWS Integrations](https://docs.spacelift.io/integrations/cloud-providers/aws#lets-explain).
- Example of Integrations created from examples/complete:
Screenshot 2024-11-18 at 14 39 48

## why

- Support all the desired Spacelift resources as code.

## references
- [internal task](https://www.notion.so/masterpoint/spacelift-aws-integration-module-12d859758a56804db291d985de4306af?pvs=4)

## Summary by CodeRabbit

- **New Features**
  - Updated the README to focus on managing Spacelift AWS integrations, including detailed usage examples and input/output parameters.
  - Introduced a new Terraform configuration file for integrating Cloud Posse's standard inputs.
  - Added a structured approach for AWS integrations, including specific configurations for each account.
  - New module and resource configurations enhance AWS integration capabilities.

- **Bug Fixes**
  - Removed outdated output variables and ensured new outputs reflect current AWS integration mappings.

- **Documentation**
  - Comprehensive updates to the README and examples to enhance clarity and usability for end-users.

- **Chores**
  - Repository name and badge link updated to reflect the new focus on Spacelift AWS integrations.
  - Updated provider configurations to ensure compatibility with the latest requirements.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,3 +1,21 @@
-resource "random_pet" "template" {
-  length = var.length
+
+module "integration_label" {
+  for_each = var.aws_integrations
+  source   = "cloudposse/label/null"
+  version  = "0.25.0"
+  name     = format("%s-%s", each.key, each.value.aws_account_id)
+  context  = module.this.context
+}
```
[the remark this answers is not in the document]
Comment (gberenice) [reply main.tf:8]: `module.this` is defined in file `context.tf`

**Review by Gowiem — APPROVED** [review]
After reading this, I'm a little worried that these child modules for spaces + aws-integration are too simple... Let's roll with them and think on it, but in reading them... all we're doing is creating some additional layer abstraction over the original resources that doesn't provide much value e.g. it doesn't provide much useful business logic and they only wrap single resources. That concerns me, but I think we go with it for now. In the future, we may want to rethink how we ship these types of resources.

File: README.md [hunk 2]
```diff
@@ -8,27 +8,31 @@ It allows to define and manage multiple AWS integrations within Spacelift, facil
...
     "prod" = {
       aws_account_id                 = "123456789012"
+      role_arn                       = "arn:aws:iam::123456789012:role/spacelift"
       duration_seconds               = 1200
       generate_credentials_in_worker = true
       labels                         = ["prod", "team:sre"]
     },
     "dev" = {
       aws_account_id                 = "210987654321"
+      role_arn                       = "arn:aws:iam::210987654321:role/spacelift"
```
[the remark this answers is not in the document]
Comment (gberenice) [reply README.md:33]: No, a random one.

**Comment by gberenice** [comment]
> After reading this, I'm a little worried that these child modules for spaces + aws-integration are too simple... Let's roll with them and think on it, but in reading them... all we're doing is creating some additional layer abstraction over the original resources that doesn't provide much value e.g. it doesn't provide much useful business logic and they only wrap single resources. That concerns me, but I think we go with it for now. In the future, we may want to rethink how we ship these types of resources.

Yeah, that's why my initial idea was to to create a subfolder with these. modules in `terraform-spacelift-automation`. I think I spent more time on writing README + examples than on actual TF code 🙈

**Comment by Gowiem** [comment]
> Yeah, that's why my initial idea was to to create a subfolder with these. modules in terraform-spacelift-automation. I think I spent more time on writing README + examples than on actual TF code 🙈

Yeah... :shipit:. Where we draw the line is hard. We have drawn it too wide this time around, but we'll come up with some better thoughts on how we deal with this in the future.

**Pull request merged**